docs/develop/typescript/failure-detection.mdx

```diff
@@ -1,164 +1,152 @@
 ---
 id: failure-detection
-title: Failure detection
+title: Failure detection - TypeScript SDK feature guide
 sidebar_label: Failure detection
-slug: /develop/typescript/failure-detection
-toc_max_heading_level: 2
+description: Learn how to set Workflow and Activity Timeouts, implement Retry Policies, and manage Activity Heartbeats using the Temporal TypeScript SDK for efficient Workflow Execution.
+toc_max_heading_level: 4
 keywords:
-  - failure-detection
+  - guide context
+  - how to
+  - typescript
+  - sdk
+  - failure detection
 tags:
+  - guide-context
+  - how-to
+  - typescript
+  - sdk
   - failure-detection
-description: Learn about Workflow and Activity timeouts, retry policies, and heartbeat settings in Temporal. Optimize Workflow Execution by configuring WorkflowExecutionTimeout, WorkflowRunTimeout, and WorkflowTaskTimeout. Understand Activity timeout types like Schedule-To-Close, Start-To-Close, and Schedule-To-Start. Use Activity Heartbeats
 ---
 
-## Workflow timeouts {#workflow-timeouts}
+This page shows how to do the following:
+
+- [Workflow Timeouts](#workflow-timeouts)
+- [Workflow retries](#workflow-retries)
+- [Activity Timeouts](#activity-timeouts)
+- [Activity Retry Policy](#activity-retries)
+- [Activity Next Retry Delay](#activity-next-retry-delay)
+- [Heartbeat an Activity](#activity-heartbeats)
+- [Activity Heartbeat Timeout](#activity-heartbeat-timeout)
+
+## Workflow Timeouts {#workflow-timeouts}
+
+**How to set Workflow Timeouts using the Temporal TypeScript SDK**
 
 Each Workflow timeout controls the maximum duration of a different aspect of a Workflow Execution.
 
-Workflow timeouts are set when [starting the Workflow Execution](#workflow-timeouts).
+Workflow Timeouts are set when [starting the Workflow Execution](#workflow-timeouts).
```

Comment:
Consider "A Client sets Workflow Timeouts when [...." because it gives this an actor vs passive voice, and also explains how those timeouts are set

Reply:
Actually, the phrase was incomplete; clarified that this is possible on both the Client and Workflow APIs, which unfortunately makes the active voice less applicable.